modules/core/src/test/java/org/apache/ignite/cache/store/StoreArrayKeyTest.java

```diff
@@ -0,0 +1,222 @@
+                .setPersistenceEnabled(true))
+        );
+
+        cfg.setCacheConfiguration(
+            new CacheConfiguration(CACHE_WITH_BACKUPS)
+                .setDataRegionName("pdr")
+                .setBackups(1)
+                .setWriteSynchronizationMode(CacheWriteSynchronizationMode.FULL_SYNC),
+            new CacheConfiguration(CACHE)
+                .setWriteSynchronizationMode(CacheWriteSynchronizationMode.FULL_SYNC)
+        );
+
+        return cfg;
+    }
+
+    /**
+     *
+     */
+    @Before
+    public void setUp() throws Exception {
+        node1 = startGrid(0);
+        node2 = startGrid(1);
+
+        node1.cluster().active(true);
+    }
+
+    /**
+     *
+     */
+    @After
+    public void tearDown() throws Exception {
+        stopAllGrids();
+        cleanPersistenceDir();
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void shouldReadWriteKey() {
+        IgniteCache<Object, Object> cache = node1.getOrCreateCache(CACHE);
+
+        cache.put(firstKey, 1);
+        cache.put(secondKey, 2);
+        cache.put(likeFirstKey, 3);
+
+        assertEquals(2, cache.size());
+
+        assertEquals(3, cache.get(likeFirstKey));
+        assertEquals(3, cache.get(firstKey));
+        assertEquals(2, cache.get(secondKey));
+
+        assertTrue(cache.containsKey(likeFirstKey));
+        assertTrue(cache.containsKey(firstKey));
+        assertTrue(cache.containsKey(secondKey));
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void shouldRemoveBySameKey() {
+        IgniteCache<Object, Object> cache = node1.getOrCreateCache(CACHE);
+
+        cache.put(firstKey, 1);
+        cache.put(secondKey, 2);
+        cache.put(likeFirstKey, 3);
+
+        assertEquals(2, cache.size());
+
+        cache.remove(firstKey);
+        cache.remove(secondKey);
+
+        assertEquals(0, cache.size());
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void shouldRemoveAllCache() {
+        IgniteCache<Object, Object> cache = node1.getOrCreateCache(CACHE);
+
+        cache.put(firstKey, "val");
+
+        cache.removeAll();
+
+        assertEquals(0, cache.size());
+
+        Iterator<Cache.Entry<Object, Object>> it = cache.iterator();
+
+        assertFalse(it.hasNext());
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void shouldClearCache() {
+        IgniteCache<Object, Object> cache = node1.getOrCreateCache(CACHE);
+
+        cache.put(firstKey, "val");
+
+        cache.clear();
+
+        assertEquals(0, cache.size());
+
+        Iterator<Cache.Entry<Object, Object>> it = cache.iterator();
+
+        assertFalse(it.hasNext());
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void shouldReadFromBackup() {
+        IgniteCache<Object, Object> cache = node2.getOrCreateCache(CACHE_WITH_BACKUPS);
+
+        cache.put(firstKey, 1);
+        cache.put(secondKey, 2);
+        cache.put(likeFirstKey, 3);
+
+        assertEquals(2, cache.size());
+
+        stopGrid(0);
```

Comment:
Why do we think that the 0th node was primary for these partitions?

Reply:
It's a mistake. I don't have any idea how to fix it. (It was removed).